### PR TITLE
Split `_updateDataset` and introduce `backendify`

### PR DESCRIPTION
**Pull request by rhuille**
- extract the pipeline preparation from `_updateDataset`, e.g. the
  dereferencing,

- introduce the `backendify` "decorator" (but can't be used as
  @backendify around functions unfortunately because JavaScript doesn't
  support it) that:

  - sets the `loading: true` property on the store at the beginning,
  - logs the error in the store if any,
  - sets the `loading: false` property on the store at the end.

### CODE REVIEW

File: playground/app.js [hunk 1]
```diff
@@ -109,6 +109,7 @@ class MongoService {
   async executePipeline(_store, pipeline, limit, offset = 0) {
     const { domain, pipeline: subpipeline } = filterOutDomain(pipeline);
     const query = mongo40translator.translate(subpipeline);
+    console.log('in executePipeline', this)
```
Comment (adimascio) [inline playground/app.js:112]: This should be removed :)

File: src/store/backend-plugin.ts [hunk 2]
```diff
@@ -87,40 +87,69 @@ export function dereferencePipelines(
   return dereferencedPipeline;
 }
 
-async function _updateDataset(store: Store<any>, service: BackendService, pipeline: Pipeline) {
-  if (!store.getters[VQBnamespace('pipeline')].length) {
-    return;
+export let backendService: BackendService; // set at plugin instantiation
+
+/**
+  `_preparePipeline` do 2 stuffs:
```
Comment (adimascio) [inline src/store/backend-plugin.ts:93]:
```suggestion
  `_preparePipeline` does 2 stuffs:
```

Comment (adimascio) [reply src/store/backend-plugin.ts:93]: I would maybe phrase it like: 

```
`_preparePipeline` responsibility is to prepare the pipeline so as to be ready for direct translation. Specifically,
this consists in 2 things:

- ...

```

File: playground/app.js (hunk 1 above)
Comment (rhuille) [reply playground/app.js]: Oupsy sorry for this

File: src/store/backend-plugin.ts (hunk 2 above)
Comment (rhuille) [reply src/store/backend-plugin.ts]: Thanks for the rephrase !

File: src/store/backend-plugin.ts [hunk 3]
```diff
@@ -87,40 +87,67 @@ export function dereferencePipelines(
...
+  return pipeline;
+}
+
+/**
+ * `backendify` is a wrapper around backend service functions that:
+ *   - sets the `loading: true` property on the store at the beginning,
+ *   - logs the error in the store if any,
+ *   - sets the `loading: false` property on the store at the end.
+ */
+export function backendify(target: Function) {
```
Comment (davinov) [inline src/store/backend-plugin.ts:117]: executeBackendRequest?
withBackendRequest? (context-manager like)

File: src/store/backend-plugin.ts [hunk 4]
```diff
@@ -87,40 +87,67 @@ export function dereferencePipelines(
...
+      const response = { error: { type: 'error', message: error.toString() } };
       store.commit(VQBnamespace('logBackendError'), {
-        backendError: { type: 'error', message: response.error },
+        backendError: response.error,
       });
-    } else {
-      store.commit(VQBnamespace('setDataset'), { dataset: response.data });
+      return response;
+    } finally {
+      store.commit(VQBnamespace('setLoading'), { isLoading: false });
```
Comment (davinov) [inline src/store/backend-plugin.ts:135]: callbacks avant requête / après succès / après erreur ?

**Pull request merged**